stdlib/os/__init__.pyi

```diff
@@ -794,11 +793,14 @@ def replace(
     src: StrOrBytesPath, dst: StrOrBytesPath, *, src_dir_fd: int | None = None, dst_dir_fd: int | None = None
 ) -> None: ...
 def rmdir(path: StrOrBytesPath, *, dir_fd: int | None = None) -> None: ...
-
-class _ScandirIterator(Iterator[DirEntry[AnyStr]], AbstractContextManager[_ScandirIterator[AnyStr], None]):
-    def __next__(self) -> DirEntry[AnyStr]: ...
-    def __exit__(self, *args: Unused) -> None: ...
-    def close(self) -> None: ...
+@final
+class _ScandirIterator(Generic[AnyStr]):
+    def __del__(self, /) -> None: ...
```

Comment:
We don't usually mark `self` as positional-only, even though it is always positional-only on C-implemented methods.

Reply:
Fair point. Reverted those changes!